subsys/portability/cmsis_rtos_v2/thread_flags.c

```diff
@@ -109,6 +109,24 @@ uint32_t osThreadFlagsWait(uint32_t flags, uint32_t options, uint32_t timeout)
 
 		time_stamp_start = (uint64_t)k_cycle_get_32();
 
+		sig = tid->signal_results & flags;
+
+		if (options & osFlagsWaitAll) {
+			/* Check if all events we are waiting on have
+			 * been signalled
+			 */
+			if (sig == flags) {
+				break;
+			}
+		} else {
+			/* Check if any events we are waiting on have
+			 * been signalled
+			 */
+			if (sig != 0) {
+				break;
+			}
+		}
+
```

Comment:
It seems a bit weird to me that this is evaluated before the call to `k_poll()`. Is the intent that it is a kind of micro-optimization so that the calling thread does not incur the system call overhead?

I worry that it might be a premature optimization in some far-away corner case.

Reply:
Sure - please see my other comment about locking.